instrumentation/active_job/lib/opentelemetry/instrumentation/active_job/instrumentation.rb

```diff
@@ -63,14 +63,17 @@ def gem_version
         end
 
         def require_dependencies
+          require 'active_support/lazy_load_hooks'
           require_relative 'patches/base'
           require_relative 'handlers'
         end
 
         def patch_activejob
-          ::ActiveJob::Base.prepend(Patches::Base) unless ::ActiveJob::Base <= Patches::Base
-
           Handlers.subscribe
```

Comment:
Should we also defer these subscriptions until an explicit  point in the Rails initialization process? Or are we assuming folks are already registering this in config/initializers themselves?

Reply:
I'm not certain this is the right approach, but when New Relic detects the presence of Rails, it will load instrumentation within a `Rails::Configuration.after_initialize` block. It would be the equivalent of wrapping [`OpenTelemetry::SDK::Configurator#install_instrumentation`](https://github.com/open-telemetry/opentelemetry-ruby/blob/672739120a605c5c14afc48e5aa07d1320443355/sdk/lib/opentelemetry/sdk/configurator.rb#L161-L168) in an `after_initialize` block. 

This could probably be done individually in instrumentation if we didn't want to tie anything about Rails into the main SDK.

Reply:
I think that's right that Instrumentation subscriptions should happen in an `after_initialize` block. I was trying to avoid adding a full-blown Railtie, but that's the fully correct way to do it.

Though from poking around Rails, it seems like simply `require "activesupport/notifications"` might be totally load-order independent. e.g.

https://github.com/rails/rails/blob/e412839f9d2e0133a50769db16709a09f87c1b7d/activerecord/lib/active_record/explain_subscriber.rb#L3-L3

Reply:
I think then we should have a follow up PR and use a Railtie. I am ok with incremental improvements.